lib/app/features/chat/providers/set_user_chat_relays_provider.c.dart

```diff
@@ -0,0 +1,40 @@
+// SPDX-License-Identifier: ice License 1.0
+
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+import 'package:ion/app/exceptions/exceptions.dart';
+import 'package:ion/app/features/chat/providers/user_chat_relays_provider.c.dart';
+import 'package:ion/app/features/nostr/providers/nostr_event_signer_provider.c.dart';
+import 'package:ion/app/features/nostr/providers/nostr_notifier.c.dart';
+import 'package:ion/app/features/user/model/user_chat_relays.c.dart';
+import 'package:ion/app/features/user/providers/user_relays_manager.c.dart';
+import 'package:riverpod_annotation/riverpod_annotation.dart';
+
+part 'set_user_chat_relays_provider.c.g.dart';
+
+///
+/// This provider is used to set the user chat relays for a given pubkey.
+/// It copies the user relays to the user chat relays and sends the event to the relay.
+///
+@riverpod
+Future<void> setUserChatRelays(
```

Comment:
In this type of provider, it is safe to call `ref.watch`

Reply:
fixed